bastislack/highline-freestyle
# Move sorting options out of the navbar

fix bastislack/highline-freestyle#231

[Link to issue](https://github.com/bastislack/highline-freestyle/issues/231)

Correct me if this way of creating a PR is not correct, never done PR to open-source before.

## What is the purpose of this PR:

Fix the Navbar responsiveness by approaching the proposed solution.

> Move the sorting options out of the navbar. Therefore the navbar becomes so short that the issue will not occur under any reasonable circumstances anymore. The sorting options could be moved next to or underneath the search bar (maybe depending on the screen size), as a drop down menu. 

## What was done to achieve 

- Created a `SearchBar` component that has both text-based filtering and sort filtering
- Since the component `SearchBar` was already created and the sort filtering would need to be implemented in the Combos page, I enhanced it with text-based filtering [(not fully implemented)](#Know limitation)
- Removed all logic of sorting from both `LeftNav` and `TopNav`
- Removed `sortOpt` state from `App.js`, it should now reduce unnecessarily re-renders 

## How to test if it works?

Open the Tricks Page and see if it looks like this:

![image](https://user-images.githubusercontent.com/78118656/225804925-c709e605-f99d-48e3-8352-24df309e4a35.png)

- Use the text search bar and check if behaves as previously.
- Use the new filter dropdown and check if behaves as previously.

Now use  `Ctrl + Shift + M` to simulate a mobile view and check if the responsiveness is good enough.

## Know limitation

I couldn't make the text-based filtering work on `ComboList.js` so I added a comment on line 30: `// TODO: Finish filtering functionality`
